tchannel/tornado/connection.py

```diff
@@ -701,3 +634,176 @@ def _request_timed_out(self, req_id, req_ttl, future):
         # while.
         future.set_exception(errors.TimeoutError())
         self._request_tombstones.add(req_id, req_ttl)
+
+
+class Reader(object):
+
+    def __init__(self, io_stream, capacity=64):
+        self.queue = tornado.queues.Queue(capacity)
+        self.filling = False
+        self.io_stream = io_stream
+
+    def fill(self):
+        self.filling = True
+
+        io_loop = IOLoop.current()
+
+        def keep_reading(f):
+            if f.exception():
+                self.filling = False
+                return log(f.exception())
+            # connect these two in the case when put blocks
+            self.queue.put(f.result())
+            io_loop.spawn_callback(self.fill)
+
+        io_loop.add_future(read_message(self.io_stream), keep_reading)
+
+    def get(self):
+        """Receive the next message off the wire.
+
+        :returns:
+            A Future that produces a Context object containing the next
```

Comment:
This is outdated. We don't have Context objects in this class anymore (I think).